docs/source/_ext/ghfiles.py

```diff
@@ -0,0 +1,65 @@
+import os.path as path
+import subprocess
+import shlex
+from sphinx.util import logging
+from docutils import nodes
+logger = logging.getLogger(__name__)
+
+
+# use an old git trick, to get the top-level, could have used ../ etc.. but
+# this will be fine..
+top = subprocess.check_output(shlex.split(
+    "git rev-parse --show-toplevel")).strip().decode("utf-8")
+
+
+def make_ref(text):
+    """ Make hyperlink to Github """
+    full_path = path.join(top, text)
+    if path.isfile(full_path):
+        ref = "https://github.com/numba/numba/blob/master/" + text
+    elif path.isdir(full_path):
+        ref = "https://www.github.com/numba/numba/tree/master/" + text
+    else:
+        logger.warn("Failed to find file in repomap: " + text)
```

Comment:
It's a shame the inverse case of "file added, repo map needs updating" is less easy to catch.

Reply:
Indeed, sigh..